src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -3,12 +3,21 @@
 import htsjdk.samtools.CigarElement;
 import htsjdk.samtools.CigarOperator;
 import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.locusiterator.AlignmentStateMachine;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
 import org.broadinstitute.hellbender.utils.read.ReadUtils;
 
-import java.util.*;
+import java.util.Deque;
+import java.util.EnumSet;
+import java.util.LinkedList;
+import java.util.List;
 
-public final class PileupElement implements Comparable<PileupElement> {
+/**
+ * Represents an individual base in a reads pileup.
+ */
+public final class PileupElement {
+    //FIXME: this static linked list is modifiable
     private final static LinkedList<CigarElement> EMPTY_LINKED_LIST = new LinkedList<>();
```

Comment:
Since you couldn't find a way to make this unmodifiable, recommend NOT declaring it as a static constant. Replace usages of `EMPTY_LINKED_LIST` with inline calls to `new LinkedList<>()`.